src/components/ChartFunctional.js

```diff
@@ -1,15 +1,13 @@
-import React,  {useState, useEffect } from 'react';
+import React from 'react';
 import { Line } from "react-chartjs-2";
 import moment from "moment";
 
-export default ({values: { balances }}) => {
+export default ({values}) => {
     let currentMonthDates = Array.from(
       {length: moment().daysInMonth() },
       (x, i) => moment().startOf("month").add(i, "days") );
-      
-    let data = () => setDataFromValues(balances.map((arr) => ({ t: arr.date, y: arr.value })));
-  
-    const [dataFromValues, setDataFromValues] = useState([23])
```

Comment:
Yo, I don't see why need of useState and useEffect was needed here...
Usually if props change in ReactJS, then component is rerendered and it will use the newest data.

Reply:
right, it works well without hooks,